pkg/api/validate/dynamic/dynamic_test.go

```diff
@@ -436,7 +436,7 @@ func TestValidateCIDRRanges(t *testing.T) {
 						{
 							ID: &workerSubnet,
 							SubnetPropertiesFormat: &mgmtnetwork.SubnetPropertiesFormat{
-								AddressPrefix: to.StringPtr("10.0.1.0/24"),
+								AddressPrefixes: to.StringSlicePtr([]string{"10.0.1.0/24"}),
```

Comment:
The logic of the flow changed, therefore the test need to change too. Would you please modify the test to cover both logical flows? I understand it is a big change, so my appologies. We need to watch for regressions in our codebase.